runtime/wasm/src/module/mod.rs

```diff
@@ -1389,6 +1392,30 @@ impl WasmInstanceContext {
         self.ctx.host_exports.log_log(&self.ctx.logger, level, msg)
     }
 
+    /// function encode(params: Array<ethereum.Value>): Bytes | null
+    fn abi_encode(
+        &mut self,
+        params_ptr: AscPtr<Array<AscPtr<AscEnum<EthereumValueKind>>>>,
+    ) -> Result<AscPtr<Uint8Array>, DeterministicHostError> {
+        let data = host_exports::abi_encode(self.asc_get(params_ptr)?);
+        // return `null` if it fails
```

Comment:
This is reasonable, but needs to be documented on the `graph-ts` api.

Reply:
You mean with a `| null` in the type annotation?

Reply:
Ah yes that's even better than just documenting it.

Reply:
I've added on `graph-ts` 🙂 
I will add later on docs!